src/main/java/org/broadinstitute/hellbender/tools/gvs/filtering/ExtractFeatures.java

```diff
@@ -75,20 +69,67 @@ public class ExtractFeatures extends ExtractTool {
         fullName = "query-labels",
         doc = "Key-value pairs to be added to the extraction BQ query. Ex: --query-labels key1=value1 --query-labels key2=value2",
         optional = true)
-
     protected List<String> queryLabels = new ArrayList<>();
 
+    @Argument(
+            fullName = "cost-observability-tablename",
+            doc = "Name of the bigquery table in which to store cost observability metadata",
+            optional = true)
+    protected String costObservabilityTableName = null;
+
+    @Argument(
+            fullName = "call-set-identifier",
+            doc = "Name of callset identifier, which is used to track costs in cost_observability table",
+            optional = true)
+    protected String callSetIdentifier = null;
+
+    @Argument(
+            fullName = "wdl-step",
+            doc = "Name of the WDL step/task (used for cost observability)",
+            optional = true)
+    protected String wdlStep = null;
+
+    @Argument(
+            fullName = "wdl-call",
+            doc = "Name of the call in the WDL step/task (used for cost observability)",
+            optional = true)
+    protected String wdlCall = null;
+
+    @Argument(
+            fullName = "shard-identifier",
+            doc = "Identifier for which shard was used in the WDL (used for cost observability)",
+            optional = true)
+    protected String shardIdentifier = null;
+
     @Override
     public boolean requiresIntervals() {
         return false;
     }
 
+    /**
+     * Enforce that if cost information is being recorded to the cost-observability-tablename then *all* recorded
+     * parameters are set
+     */
+    @Override
+    protected String[] customCommandLineValidation() {
+        final List<String> errors = new ArrayList<>();
+        if (projectID != null || datasetID != null || costObservabilityTableName != null || callSetIdentifier != null || wdlStep != null || wdlCall != null || shardIdentifier != null) {
+            if (projectID == null || datasetID == null || costObservabilityTableName == null || callSetIdentifier == null || wdlStep == null || wdlCall == null || shardIdentifier == null) {
+                errors.add("Parameters 'project-id', 'dataset-id', `cost-observability-tablename`, `call-set-identifier`, 'wdl-step', 'wdl-call', and 'shardIdentifier' must either ALL be set or ALL NOT BE set");
```

Comment:
here too

Reply:
same — doesn't look like the params were treated in a consistent way